modules/host/host.go

```diff
@@ -105,8 +106,19 @@ func New(cs *consensus.ConsensusSet, hdb modules.HostDB, tpool modules.Transacti
 	if err != nil {
 		return nil, err
 	}
+	host, err := modules.IGD.ExternalIP()
+	if err != nil {
+		// log error? should be fatal
+		host = "::1"
+	}
 	_, port, _ := net.SplitHostPort(h.listener.Addr().String())
-	h.myAddr = modules.NetAddress(net.JoinHostPort(modules.ExternalIP, port))
+	h.myAddr = modules.NetAddress(net.JoinHostPort(host, port))
+
+	// Forward port, if possible
+	portInt, _ := strconv.Atoi(port)
+	if portErr := modules.IGD.Forward(uint16(portInt), "Sia Host"); portErr != nil {
+		// log error? should be fatal if err != modules.ErrNoUPnP
```

Comment:
if you're expecting your port to be forwarded and it isn't, that's definitely a problem.
Then you run into another issue though, which is when users forwarded their ports manually. Then they shouldn't care if UPnP fails, but if it does it will return an error. So then we need a `--no-upnp` flag to disable the check or something. Or we just fail silently/log it in all cases. But then we need a way to inform the user whether or not they're reachable, which so far has had its own bugs...